lms/djangoapps/program_enrollments/api/linking.py

```diff
@@ -145,6 +171,39 @@ def _get_lms_users(lms_usernames):
     }
 
 
+def _unlink_course_enrollments_from_program_course_enrollment(program_enrollment):
+    """
+    Unlinks CourseEnrollments from the ProgramEnrollment by doing the following for
+    each ProgramCourseEnrollment associated with the Program Enrollment.
+        1. unenrolling the corresponding user from the course
+        2. moving the user into the audit track
+        3. removing the link between the ProgramCourseEnrollment and the CourseEnrollment
+
+    Note: It is unlikely that this method should be used on its own.
+    It should only be used as part of the larger unlinking process for a learner via the support tool.
+
+    Arguments:
+        program_enrollment: the ProgramEnrollment object
+    """
+    program_course_enrollments = program_enrollment.program_course_enrollments.all()
```

Comment:
TODO: I will create a ticket to include logic around handling the transference of `course_staff` roles.

Reply:
See [MST-238](https://openedx.atlassian.net/browse/MST-238).